### PR TITLE
fix/dialog errors in edit/delete

### PR DESCRIPTION
**Pull request by syrok94**
**What does this PR do?**
fixed persistence error in edit and dialog.

**Related Issue(s)**
none

**Changes**
List the changes made in this PR. This could include new features, bug fixes, improvements, or any other modifications.
reset setError.

**Screenshots or GIFs (if applicable)**
none

**Additional Comments**
none

**By submitting this PR, I confirm that:**
- [x] I have reviewed my code and believe it is ready for merging.
- [x] I understand that this PR may be subject to review and changes.
- [x] I agree to abide by the code of conduct and contributing guidelines of this project.

### CODE REVIEW

**Review by bryanlundberg — COMMENTED** [review]
Works great, but it's the same code at the end. Can you improve it, refactoring as a custom hook for both cases?

**Comment by syrok94** [comment]
Hi @bryanlundberg, I have created a custom hook, useResetError. Please review it once and let me know if any further changes are required.

File: src/components/dialogs/dialog-delete-collection/dialog-delete-collection.tsx [hunk 1]
```diff
@@ -19,12 +20,16 @@ import { useState } from "react";
 export default function DialogDeleteCollection() {
   const t = useTranslations("Index");
   const { setCubes } = useTimerStore();
-  const { cube, closeDialog } = useDialogCubesOptions();
+  const { cube, closeDialog , isOpen } = useDialogCubesOptions();
   const [cubeName, setCubeName] = useState("");
   const [error, setError] = useState({
     status: false,
     message: "",
   });
```
Comment (bryanlundberg) [inline src/components/dialogs/dialog-delete-collection/dialog-delete-collection.tsx:28]: Moving this state inside hook

File: src/hooks/useErrorReset.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { useEffect } from "react";
+
+export default function useErrorReset(isOpen:boolean , setError: React.Dispatch<React.SetStateAction<{ status: boolean; message: string }>>) {
+  useEffect(() => {
+    if(!isOpen){
+        setError({ status: false, message: "" });
+    }
+  },[isOpen , setError]);
+}
```
Comment (bryanlundberg) [inline src/hooks/useErrorReset.ts:9]: Here you would need to include the error useState, also return the methods to control the hook outside.

File: src/components/dialogs/dialog-edit-collection/dialog-edit-collection.tsx [hunk 3]
```diff
@@ -92,6 +93,9 @@ export default function DialogEditCollection() {
     }
   };
 
+  // helps to reset error state
+  useErrorReset(isOpen , setError);
```
Comment (bryanlundberg) [inline src/components/dialogs/dialog-edit-collection/dialog-edit-collection.tsx:97]: With the destructuring of the states can be more clean.

Not exactly, but this is a little bit the idea:
`const { status, message, setError, resetError } = useErrorReset();`

**Review by bryanlundberg — COMMENTED** [review]
Hi, I have some minor tips to simplify the implementation. 👋

**Comment by bryanlundberg** [comment]
Related issue #345

File: src/components/dialogs/dialog-edit-collection/dialog-edit-collection.tsx (hunk 3 above)
Comment (syrok94) [reply src/components/dialogs/dialog-edit-collection/dialog-edit-collection.tsx:97]: Sure, will do it this way.

Comment (bryanlundberg) [reply src/components/dialogs/dialog-edit-collection/dialog-edit-collection.tsx:97]: How is it going with this task ? @syrok94 😅

Comment (syrok94) [reply src/components/dialogs/dialog-edit-collection/dialog-edit-collection.tsx:97]: Hii @bryanlundberg , sorry for the late replies , I tried to put error useState in the hook but it keeps on showing infinite render error. Also, I am quite busy so unable to give my best.

Comment (bryanlundberg) [reply src/components/dialogs/dialog-edit-collection/dialog-edit-collection.tsx]: Don't worry, I can continue the job from here 😅

**Review by bryanlundberg — APPROVED** [review]

**Pull request merged**